### ISSUE TITLE
watcher noble need fix incompatiablity between apscheduler and eventlet

### ISSUE THREAD

**Issue by jneo8** [canonical/ubuntu-openstack-rocks#75]
## Description
The current version of Watcher in the **Noble** release encounters compatibility issues between **APScheduler** and **eventlet**, leading to runtime failures.

A fix addressing this incompatibility has been merged into Watcher's master branch as of **2024-12-03**:  
[Fix Merge Request](https://review.opendev.org/c/openstack/watcher/+/933890)

## Current Version in image

The version of Watcher currently used in the image is as follows:

```plaintext
python3-watcher/noble,now 2:12.0.0-0ubuntu1 all [installed,automatic]
watcher-api/noble,now 2:12.0.0-0ubuntu1 all [installed]
watcher-applier/noble,now 2:12.0.0-0ubuntu1 all [installed]
watcher-common/noble,now 2:12.0.0-0ubuntu1 all [installed,automatic]
watcher-decision-engine/noble,now 2:12.0.0-0ubuntu1 all [installed]
```

## Possible solution

The temporary workaround can be downgrade the base of watcher from 24.04 to 22.04, which python is no longer 3.12 so the issue won't happen.

The core solution is to perform an [SRU](https://canonical-sru-docs.readthedocs-hosted.com/en/latest/) for the watcher package.

**Comment by jneo8** [comment]
The workaround will have the issue that watcher is using deprecated metric `cpu_util`, which is been fixed on noble release.

**Comment by hemanthnakkina** [comment]
@jneo8 https://github.com/canonical/ubuntu-openstack-rocks/pull/76 is merged now and rock is downgraded to 22.04

Is this is `cpu_util` PR https://review.opendev.org/c/openstack/watcher/+/898791 you are mentioning, if so it should be available on watcher 12.0.0 and so should be available in the rock.

**Comment by jneo8** [comment]
Yes, so the PR do works for me and I already verify it's working. 

But should we close this issue now or should we close it after we make sure it's using ubuntu 24.04 with watcher noble release back port to support python3.12?

**Comment by hemanthnakkina** [comment]
If the newly built rock watcher:2024.1 with base as 22.04 works with no issue, we can close the issue.
